jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/ProgressSpinnerTests.java

```diff
@@ -0,0 +1,123 @@
+    public void checkSpinnerAppearAndThenDisappear() {
+        showSpinner.click();
+        basicProgressSpinner.is().displayed();
+        waitCondition(() -> basicProgressSpinner.isHidden());
+    }
+
+    @Test(description = "Test checks that progress spinner disappears after being displayed")
+    public void disappear() {
+        showSpinner.click();
+        new Timer(6000L).wait(() -> basicProgressSpinner.is().disappear());
+    }
+
+    @Test(description = "Test checks an interval during which spinner is displayed")
+    public void checkSpinnerDurationTest() {
+        refresh();
+        showSpinner.show();
+        final int DURATION = 5;
+        JAction action = () -> {
+            basicProgressSpinner.base().timer().wait(() -> basicProgressSpinner.isDisplayed());
+            basicProgressSpinner.base().timer().wait(() -> basicProgressSpinner.isHidden());
+        };
+        showSpinner.click();
+        duration(DURATION, 900, action);
+    }
+
+    @Test(description = "Test checks configurable progress spinner")
+    public void configurableSpinnerBasicTest() throws Exception {
+        configuredSpinner.shouldBe().displayed();
+        configuredSpinner.show();
+        configuredSpinner.shouldBe().visible();
+        configuredSpinner.has().mode(DETERMINATE);
+        configuredSpinner.has().value(50);
+        configuredSpinner.has().color(DEEP_PURPLE_2.value());
+        configuredSpinner.has().diameter("100px");
+        configuredSpinner.has().strokeWidth("10%");
+    }
+
+    @Test(description = "Test checks progress spinner's color attribute")
+    public void checkConfigurableSpinnerColorTest() {
+        configuredSpinner.show();
+        configuredSpinner.has().color(DEEP_PURPLE_2.value());
+        progressSpinnerAccentColorRadio.click();
+        configuredSpinner.has().color(AMBER_ACCENT_5.value());
+        progressSpinnerWarnColorRadio.click();
+        configuredSpinner.has().color(RED_2.value());
+        progressSpinnerIndeterminateModeRadio.click();
+        configuredSpinner.has().mode(INDETERMINATE);
+        configuredSpinner.has().color(RED_2.value());
+        progressSpinnerPrimaryColorRadio.click();
+        waitCondition(() -> configuredSpinner.color().equals(DEEP_PURPLE_2.value()));
+        configuredSpinner.has().color(DEEP_PURPLE_2.value());
+        progressSpinnerDeterminateModeRadio.click();
+    }
+
+    @Test(description = "Test checks progress spinner's mode attribute")
+    public void checkConfigurableSpinnerModeTest() {
+        configuredSpinner.show();
+        configuredSpinner.has().mode(DETERMINATE);
+        progressSpinnerIndeterminateModeRadio.click();
+        configuredSpinner.has().mode(INDETERMINATE);
+        progressSpinnerDeterminateModeRadio.click();
+        configuredSpinner.has().mode(DETERMINATE);
+    }
+
+    @Test(description = "Test checks progress spinner's values transformation")
+    public void checkConfigurableSpinnerValueTest() throws Exception {
```

Comment:
Please, avoid using generic exception handling: In the `configurableSpinnerBasicTest()` method, there is a generic `throws Exception` clause in the method signature. It is generally recommended to handle specific exceptions instead of using generic exception handling.

Reply:
Fixed